src/analyze/test-verify.c

```diff
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#include "analyze-verify.h"
+#include "tests.h"
+
+static void test_verify_nonexistent(void) {
+        /* Negative cases */
+        assert_se(verify_executable(NULL, &(ExecCommand) {.flags = EXEC_COMMAND_IGNORE_FAILURE, .path = (char*) "/non/existant"}) == 0);
+        assert_se(verify_executable(NULL, &(ExecCommand) {.path = (char*) "/non/existant"}) < 0);
```

Comment:
exist**e**nt

...and if verify_executable takes a const*, then this ugly cast to `char*` will not be necessary.

Reply:
Is that true? `ExecCommand` might be const but the type of `path` is still `char*` hence we get:
```
../src/analyze/test-verify.c:12:105: error: initializing 'char *' with an expression of type 'const char [10]' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
        assert_se(verify_executable(NULL, &(ExecCommand) {.flags = EXEC_COMMAND_IGNORE_FAILURE, .path = "/bin/echo"}) == 0);
```

I'm afraid that it's either:
* this cast;
* or converting `path` to be `const char *` but then this PR would touch a completely unrelated part as well.

Unless I am missing something. Thoughts?

Reply:
Yeah, using const in various places would be nice, but it would require changing many different spots.